Sources/SCNPath/RKPathEntity.swift

```diff
@@ -0,0 +1,215 @@
+        rectangleLocal.look(at: newPosition, from: lastPosition, relativeTo: realityKitPath.anchor!)
+        
+        //Put a circle at the tip of the rectangle.
+        let circleLocal = circleEntity!.clone(recursive: true)
+        realityKitPath.anchor!.addChild(circleLocal)
+        circleLocal.setPosition(newPosition, relativeTo: nil)
+        pathEntitiesInScene.append(circleLocal)
+        
+
+    }
+        
+        
+        
+        
+   // override func recalcGeometry() {
+        //super.recalcGeometry()
+
+        //if we didn't use a different file path each time, the model would Not change.
+        //makeRKGeo(path: "rkPath\(entityCount).usdz")
+        //entityCount += 1
+        
+        //Delete the old usdz files we saved to the device.
+        //let appDelegate = UIApplication.shared.delegate as! AppDelegate
+        //appDelegate.deleteTemporaryDirectory()
+    //}
+    
+    
+    ///Make a .usdz file from the SceneKit scene that includes the SceneKit path so that we can use it in RealitKit.
+    private func makeRKGeo(path : String){
+        let url = FileManager.default.temporaryDirectory
+        let finalURL = url.appendingPathComponent(path)
+        scene?.write(to: finalURL, delegate: nil, progressHandler: {
+            completion,error,unsafe  in
+            print("completion:", completion)
+        })
+        loadEntity(fromURL: finalURL)
+    }
+        
+    ///Load the .usdz file as a RealityKit Entity
+    private func loadEntity(fromURL url: URL){
+        var cancellable: AnyCancellable? = nil
+        cancellable = Entity.loadAsync(contentsOf: url)
+            .sink(receiveCompletion: { completion in
+                cancellable?.cancel()
+            }, receiveValue: { (loadedEntity: Entity) in
+                self.didLoadNewUSDZModel(model: loadedEntity)
+                cancellable?.cancel()
+            })
+    }
+    
+    
+    ///When the new .usdz file has loaded, update realityKitPath to have the same mesh and material.
+    private func didLoadNewUSDZModel(model: Entity) {
+        print("didLoadNewUSDZModel", model)
+        
+        //The mesh and material of the rkPath now update to be the same as the generated SCN geometry.
+        self.realityKitPath.components[ModelComponent.self] = model.findModelEntity()?.model
+    }
+    
+    
+    
+    
+    
+    
+    
+}
+
+
+extension SCNVector3 {
+    func getSIMD3Float() -> SIMD3<Float>{
+        return SIMD3<Float>([self.x,
+                             self.y,
+                             self.z])
+    }
+}
+
+
+
+extension Entity {
+    ///From Reality Composer, the visible ModelEntities are children of nonVisible Entities
+    ///Recursively searches through all descendants for a ModelEntity, Not just through the direct children.
+    ///Reutrns the first model entity it finds.
+    ///Returns the input entity if it is a model entity.
+    func findModelEntity() -> ModelEntity? {
```

Comment:
I'm surprised this works, might be better to just replace all instances of ModelEntity with HasModel, as sometimes when importing models they do not correlate - are you not seeing an issue with this one?

Reply:
I have never had an issue with it but yes it is a good idea to change it to HasModel so I'll do that.